src/Features/Core/Shared/Utilities/PatternMatcher.cs

```diff
@@ -165,9 +165,19 @@ public PatternMatcher(string pattern, CultureInfo culture, bool verbatimIdentifi
             _compareInfo = culture.CompareInfo;
 
             _fullPatternSegment = new Segment(pattern, verbatimIdentifierPrefixIsWordCharacter);
-            _dotSeparatedSegments = pattern.Split(s_dotCharacterArray, StringSplitOptions.RemoveEmptyEntries)
-                                            .Select(text => new Segment(text.Trim(), verbatimIdentifierPrefixIsWordCharacter))
-                                            .ToArray();
+
+            if (pattern.IndexOf('.') < 0)
+            {
+                // PERF: Avoid string.Split allocations when the pattern doesn't contain a dot.
+                _dotSeparatedSegments = pattern.Length > 0 ? new Segment[1] { _fullPatternSegment } : SpecializedCollections.EmptyArray<Segment>();
+            }
+            else
+            {
+                _dotSeparatedSegments = pattern.Split(s_dotCharacterArray, StringSplitOptions.RemoveEmptyEntries)
```

Comment:
We do plan to do better here, I would hope we would get to the place where we have no unnecessary allocations in the case where you don't actually have to split.  We would probably always return a new empty array however, instead of interning to Array.Empty<T>, like you do here, for compat reasons.